recipes/googleapis/all/conanfile.py

```diff
@@ -11,18 +17,135 @@ class GoogleAPIS(ConanFile):
     homepage = "https://github.com/googleapis/googleapis"
     topics = "google", "protos", "api"
     settings = "os", "arch", "compiler", "build_type"
+    generators = "cmake", "cmake_find_package_multi"
+    options = {
+        "shared": [True, False], 
+        "fPIC": [True, False]
+        }
+    default_options = {
+        "shared": False, 
+        "fPIC": True
+        }
+    exports = "helpers.py"
     short_paths = True
 
+    def export_sources(self):
+        self.copy("CMakeLists.txt")
+
     def source(self):
         get(self, **self.conan_data["sources"][str(self.version)], destination=self.source_folder, strip_root=True)
 
-    def package_id(self):
-        self.info.header_only()
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+            self.options["protobuf"].shared = True
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, 11)
+        if self.settings.compiler == "gcc" and tools.Version(self.settings.compiler.version) <= "5":
+            raise ConanInvalidConfiguration("Build with GCC 5 fails")
+
+        if self.settings.compiler in ["Visual Studio", "msvc"] and self.options.shared:
+            raise ConanInvalidConfiguration("Source code generated from protos is missing some export macro")
```

Comment:
This can probably be fixed by patching generated files to export a couple of structs.